eclipse/src/saros/filesystem/EclipseReferencePointManager.java

```diff
@@ -0,0 +1,215 @@
+   *
+   * @param resource for which the reference point should be created
+   * @return the reference point of given resourcen
+   */
+  public static IReferencePoint create(IResource resource) {
+    if (resource == null) return null;
+
+    IProject project = resource.getProject();
+    saros.filesystem.IPath path = ResourceAdapterFactory.create(project.getFullPath());
+
+    return new ReferencePointImpl(path);
+  }
+
+  /**
+   * Inserts a pair of {@link IReferencePoint} reference point and {@link IProject} project. The
+   * reference point will be created from the project's relative path by
+   * {#putIfAbsent(IReferencePoint referencePoint, IProject project) putIfAbsent} .
+   *
+   * @param project the {@link IProject} project for which a reference point is created and are
+   *     inserted.
+   */
+  public void putIfAbsent(IProject project) {
+    IReferencePoint referencePoint = create(project);
+
+    putIfAbsent(referencePoint, project);
+  }
+
+  /**
+   * Insert a pair of {@link IReferencePoint} reference point and {@link IProject} project
+   *
+   * @param referencePoint the key of the pair
+   * @param project the value of the pair
+   */
+  public void putIfAbsent(IReferencePoint referencePoint, IProject project) {
+    if (!referencePointToProjectMapper.containsKey(referencePoint)) {
+      referencePointToProjectMapper.put(referencePoint, project);
+    }
+  }
+
+  /**
+   * Returns the {@link IProject} project given by the {@link IReferencePoint} reference point
+   *
+   * @param referencePoint the key for which the IProject should be returned
+   * @return the IProject given by referencePoint
+   */
+  public org.eclipse.core.resources.IProject getProject(IReferencePoint referencePoint) {
+    return referencePointToProjectMapper.get(referencePoint);
+  }
+
+  /**
+   * Returns the {@link IResource} resource in combination of the {@link IReferencePoint} reference
+   * point and the {@link IPath} relative path from the {@link IReferencePoint} reference point to
+   * the {@link IResource} resource, or null if the resource does not exist
+   *
+   * @param referencePoint The {@link IReferencePoint} reference point, on which the {@link
+   *     IResource} resource belongs to
+   * @param referencePointRelativePath the {@link IPath} relative path from the {@link
+   *     IReferencePoint} reference point to the {@link IResource} resource
+   * @return the resource of the {@link IReferencePoint} reference point from {@link IPath}
+   *     referencePointRelativePath or null if resource does not exist.
+   */
+  public IResource findMember(IReferencePoint referencePoint, IPath referencePointRelativePath) {
+    if (referencePoint == null) throw new IllegalArgumentException("Reference point is null");
+
+    if (referencePointRelativePath == null) throw new IllegalArgumentException("Path is null");
+
+    org.eclipse.core.resources.IProject project = getProject(referencePoint);
```

Comment:
as a suggestion you could remove the followup NPE check and use `Objects.requireNonNull`
```suggestion
    org.eclipse.core.resources.IProject project = Objects.requireNonNull(getProject(referencePoint),
     "No project found for reference point: " + referencePoint);
```

Reply:
This is excellent. Thank you!